store/migration/10001__init_schema.sql

```diff
@@ -59,6 +59,63 @@ VALUES
         ''
     );
 
+-- 3rd parties auth provider
+CREATE TABLE third_party_auth_provider
+(
+    id         INTEGER PRIMARY KEY AUTOINCREMENT,
+    row_status TEXT    NOT NULL CHECK (
+        row_status IN ('NORMAL', 'ARCHIVED')
+        )                       DEFAULT 'NORMAL',
+    creator_id INTEGER NOT NULL REFERENCES principal (id),
+    created_ts BIGINT  NOT NULL DEFAULT (strftime('%s', 'now')),
+    updater_id INTEGER NOT NULL REFERENCES principal (id),
+    updated_ts BIGINT  NOT NULL DEFAULT (strftime('%s', 'now')),
+    `type`     TEXT    NOT NULL CHECK (`type` LIKE 'bb.auth.provider.%'),
+    payload    TEXT    NOT NULL
+);
+
+CREATE TRIGGER IF NOT EXISTS `trigger_update_third_party_auth_provider_modification_time`
+    AFTER
+        UPDATE
+    ON `third_party_auth_provider`
+    FOR EACH ROW
+BEGIN
+    UPDATE
+        `third_party_auth_provider`
+    SET updated_ts = (strftime('%s', 'now'))
+    WHERE rowid = old.rowid;
+END;
+
+-- 3rd parties account
+CREATE TABLE third_party_account
```

Comment:
would be possible to reuse the existing principle table?

I am thinking we may need to add an auth_provide column there. The enum is [BYTEBASE, GITLAB_SELF_HOST]

Reply:
If we adopt this way, I think we may have to do a little refactor:
Bytebase's account is a concept different from other auth providers' accounts, say google.
If we store them in a single table, maybe we should adopt a payload field as well as a provider field for multiple auth providers.

Reply:
If we find we need extra info for GitLab feature, I think payload is the way to go.

We only want to keep a single representation of a user (the principal table) otherwise, membership and other stuff will be quite complicated (we have a lot of stuff related with the principal_id).